### PR TITLE
collect_predictions() for survival models

### PR DESCRIPTION
**Pull request by topepo**
Includes test cases for tidymodels/tune#798

### CODE REVIEW

File: .github/workflows/GH-R-CMD-check.yaml [hunk 1]
```diff
@@ -76,7 +76,7 @@ jobs:
           try(pak::pkg_install("tidymodels/stacks"))
           try(pak::pkg_install("tidymodels/themis"))
           try(pak::pkg_install("tidymodels/tidymodels"))
-          try(pak::pkg_install("tidymodels/tune"))
+          try(pak::pkg_install("tidymodels/tune#798"))
```
Comment (hfrick) [inline .github/workflows/GH-R-CMD-check.yaml:79]: TODO before merging this PR: point back to `main`

File: tests/testthat/test-survival-tune_race_anova.R [hunk 2]
```diff
@@ -241,6 +276,45 @@ test_that("race tuning (anova) survival models with integrated metric", {
...
+  expect_equal(unsum_pred[0,], integrated_ptype)
+  expect_equal(nrow(unsum_pred), integrated_oob * nrow(aov_finished))
+
+  expect_equal(unsum_pred$.pred[[1]][0,], integrated_list_ptype)
+  expect_equal(nrow(unsum_pred$.pred[[1]]), length(time_points))
+
+
+  sum_pred <- collect_predictions(aov_integrated_res, summarize = TRUE)
+  expect_equal(sum_pred[0,], integrated_ptype[, names(integrated_ptype) != "id"])
+  expect_equal(nrow(sum_pred), 728)
```
Comment (hfrick) [inline tests/testthat/test-survival-tune_race_anova.R:314]: where does the `728` come from? I'm struggling to reproduce this locally.

Comment (hfrick) [reply tests/testthat/test-survival-tune_race_anova.R:314]: FYI also `expect_equal(nrow(sum_pred), nrow(sim_tr) * nrow(aov_finished))` fails for this one locally.

File: tests/testthat/_snaps/survival-tune_race_win_loss.md [hunk 3]
```diff
@@ -1,15 +1,31 @@
+# race tuning (win_loss) survival models with integrated metric
+
+    Evaluation times are only required when dynmanic or integrated metrics are selected as the primary metric (and will be ignored).
+
+---
+
+    Evaluation times are only required when dynmanic or integrated metrics are selected as the primary metric (and will be ignored).
```
Comment (hfrick) [inline tests/testthat/_snaps/survival-tune_race_win_loss.md:7]: same as for `tune_race_anova()`

File: tests/testthat/_snaps/survival-tune-sa.md [hunk 4]
```diff
@@ -1,42 +1,52 @@
...
 
 # sim annealing tuning survival models with mixture of metric types
 
-    No evaluation time was set; a value of 10 was used.
+    4 evaluation times were specified during tuning; the first (10) will be used.
 
 ---
 
-    No evaluation time was set; a value of 10 was used.
+    No evaluation time was set; a value of 5 was used.
```
[the remark this answers is not in the document]
Comment (hfrick) [reply tests/testthat/_snaps/survival-tune-sa.md:19]: On a second look, I think I might have overlooked the `set.seed()` line in the previous version, i.e., we are not newly setting a seed, and thus this change and some of the other changes below with the same comment may be worth looking into further.

File: tests/testthat/_snaps/survival-tune-bayes.md [hunk 5]
```diff
@@ -1,15 +1,27 @@
+# Bayesian tuning survival models with integrated metric
+
+    Evaluation times are only required when dynmanic or integrated metrics are selected as the primary metric (and will be ignored).
```
[the remark this answers is not in the document]
Comment (hfrick) [reply tests/testthat/_snaps/survival-tune-bayes.md:3]: Now logged at https://github.com/tidymodels/tune/issues/802

[the remark this answers is not in the document]
Comment (topepo) [reply tests/testthat/_snaps/survival-tune-bayes.md:3]: https://github.com/tidymodels/tune/issues/802

File: tests/testthat/_snaps/survival-tune-bayes.md [hunk 6]
```diff
@@ -1,15 +1,27 @@
...
+
 # Bayesian tuning survival models with dynamic metric
 
+    4 evaluation times were specified during tuning; the first (10) will be used.
```
[the remark this answers is not in the document]
Comment (topepo) [reply tests/testthat/_snaps/survival-tune-bayes.md:7]: How about: 

>  4 evaluation times were available; the first (10) will be used.

File: tests/testthat/_snaps/survival-tune-sa.md [hunk 7]
```diff
@@ -1,42 +1,52 @@
+# sim annealing tuning survival models with integrated metric
+
+    Evaluation times are only required when dynmanic or integrated metrics are selected as the primary metric (and will be ignored).
```
[the remark this answers is not in the document]
Comment (topepo) [reply tests/testthat/_snaps/survival-tune-sa.md:3]: Also https://github.com/tidymodels/tune/issues/802

File: tests/testthat/test-survival-tune_race_anova.R [hunk 8]
```diff
@@ -5,7 +5,7 @@ suppressPackageStartupMessages(library(finetune))
 skip_if_not_installed("finetune", minimum_version = "1.1.0.9001")
 skip_if_not_installed("parsnip", minimum_version = "1.1.0.9003")
 skip_if_not_installed("censored", minimum_version = "0.2.0.9000")
-skip_if_not_installed("tune", minimum_version = "1.1.1.9001")
+skip_if_not_installed("tune", minimum_version = "1.1.2.9009")
 skip_if_not_installed("yardstick", minimum_version = "1.2.0.9001")
 skip_if_not_installed("finetune", minimum_version = "1.1.0.9002")
```
Comment (topepo) [inline tests/testthat/test-survival-tune_race_anova.R:10]:
```suggestion
skip_if_not_installed("finetune", minimum_version = "1.1.0.9004")
```

and update the other file(s) that use finetune

File: tests/testthat/test-survival-tune_race_anova.R (hunk 2 above)
Comment (topepo) [reply tests/testthat/test-survival-tune_race_anova.R:312]: So I figured it out. I did a low number of resamples (6) to save computing time. With this small number, not all of the training set points are in at least one of the OOB assessment sets.  When you multiply this by the number of parameters that finish the race, you get the right number. I'll update the logic. 

```
> num_hold_out_rows <- 
+     sim_rs %>% 
+     mutate(
+         oob_rows = map(splits, ~ as.integer(.x, data = "assessment"))
+     )  %>% 
+     pluck("oob_rows") %>% 
+     unlist() %>% 
+     unique() %>% 
+     length()
> num_hold_out_rows
[1] 364
> nrow(sim_tr)
[1] 375
> num_hold_out_rows * nrow(aov_finished)
[1] 728
```

File: .github/workflows/GH-R-CMD-check.yaml (hunk 1 above)
Comment (topepo) [reply .github/workflows/GH-R-CMD-check.yaml]: 798 is merged!

Comment (hfrick) [reply .github/workflows/GH-R-CMD-check.yaml]: yep, I've updated it to use tune#804 until we're done with that!

File: tests/testthat/_snaps/survival-tune_race_anova.md [hunk 9]
```diff
@@ -1,6 +1,6 @@
 # race tuning (anova) survival models with integrated metric
 
-    Evaluation times are only required when dynmanic or integrated metrics are selected as the primary metric (and will be ignored).
+    An evaluation time is only required when a dynamic metric is selected (and `eval_time` will thus be ignored).
```
Comment (hfrick) [inline tests/testthat/_snaps/survival-tune_race_anova.md:3]: Can we add the call to this warning?

File: tests/testthat/_snaps/survival-tune-bayes.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (topepo) [reply tests/testthat/_snaps/survival-tune-bayes.md]: Changed in [0cef69c](https://github.com/tidymodels/extratests/pull/158/commits/0cef69c0f6158877554ddc8064cec7220b078ad4)

File: tests/testthat/_snaps/survival-tune_race_anova.md (hunk 9 above)
Comment (topepo) [reply tests/testthat/_snaps/survival-tune_race_anova.md:3]: changes in tidymodels/tune#806; tests updates here in a few min

Comment (topepo) [reply tests/testthat/_snaps/survival-tune_race_anova.md]: updated snapshots in [43f2cd1](https://github.com/tidymodels/extratests/pull/158/commits/43f2cd11d5e5e00e2f415eec81706f52aa260294)
